src/util/TermUtil.ts

```diff
@@ -40,8 +40,8 @@ export const isTerm = (input?: any): input is Term =>
  * Converts a subject to a named node when needed.
  * @param subject - Subject to potentially transform.
  */
-export const toSubjectTerm = (subject: NamedNode | string): NamedNode =>
-  typeof subject === 'string' ? namedNode(subject) : subject;
+export const toSubjectTerm = <T extends NamedNode | BlankNode>(subject: T | string): T =>
+  typeof subject === 'string' ? namedNode(subject) as any : subject;
```

Comment:
Unfortunately, ESLint's `func-style` prevents me from typing this more accurately: https://github.com/typescript-eslint/typescript-eslint/issues/1236

Would it be an idea to set the `func-style` rule to `declaration` instead? (I like that better anyway.)

Reply:
If I had overloads, I could type it correctly:

```
function toSubjectTerm(subject: string): NamedNode;
function toSubjectTerm<T extends Term>: T;
```

With `T | NamedNode`, we lose the specific typing. But not that bad.
Note however that `toSubjectTerm` just mimics `toObjectTerm` below it.

Reply:
Migrated to `declaration` in f9a20799eb4a8a211ef1316f49ee3cedad0187e7.